plugins/providers/oss/provider.go

```diff
@@ -317,12 +355,21 @@ func updatePolicyToGrantPermissions(policy string, g domain.Grant) (string, erro
 		return "", err
 	}
 
-	statements, matchingStatements := findStatementsWithMatchingActions(bucketPolicy, resourceAccountID, g)
+	statements, matchingStatements, additionalMatchingStatements := findStatementsWithMatchingActions(bucketPolicy, resourceAccountID, g)
+
+	resource := fmt.Sprintf("acs:oss:*:%s:%s", resourceAccountID, g.Resource.URN)
+	resourceWithWildcard := fmt.Sprintf("acs:oss:*:%s:%s/*", resourceAccountID, g.Resource.URN)
+	resources := []string{resourceWithWildcard}
+
+	if g.Role == "admin" {
```

Comment:
we can't rely on g.Role as this would be defined by the user in the provider config, technically user would be able to define the role_id with anything other than "admin"